### PR TITLE
prov/sm2: Add util srx support to SM2

### PR DESCRIPTION
**Pull request by a-szegel**
sm2 has internal receive queues that can now be replaced with the generic util_srx

This also adds the missing progress functions needed for sm2 to work with the util_srx progress support

### CODE REVIEW

File: prov/sm2/src/sm2.h [hunk 1]
```diff
@@ -177,48 +183,7 @@ struct sm2_domain {
...
-	struct sm2_queue unexp_msg_queue;
-	struct sm2_queue unexp_tagged_queue;
-	struct sm2_recv_fs *recv_fs;
-
-	// TODO Determine if this spin lock is needed.
-	ofi_spin_t lock;
-};
-
-struct sm2_rx_entry *sm2_alloc_rx_entry(struct sm2_srx_ctx *srx);
+OFI_DECLARE_FREESTACK(struct sm2_xfer_ctx, sm2_xfer_ctx_fs);
```
Comment (aingerson) [inline prov/sm2/src/sm2.h]: Drop

File: prov/sm2/src/sm2_ep.c [hunk 2]
```diff
@@ -454,6 +319,18 @@ static int sm2_ep_close(struct fid *fid)
...
+int sm2_progress_addr(struct fi_peer_rx_entry *rx_entry)
+{
+	struct sm2_xfer_ctx *xfer_ctx = rx_entry->peer_context;
+	struct sm2_ep *ep = xfer_ctx->ep;
+	struct sm2_av *sm2_av =
+		container_of(ep->util_ep.av, struct sm2_av, util_av);
+
+	rx_entry->addr =
+		sm2_av->reverse_lookup[xfer_ctx->xfer_entry.sender_gid];
+	return FI_SUCCESS;
```
Comment (aingerson) [inline prov/sm2/src/sm2_ep.c]: return success only if the address is now valid. see shm

**Review by aingerson — COMMENTED** [review]
2 things, otherwise this looks correct!

File: prov/sm2/src/sm2.h (hunk 1 above)
Comment (a-szegel) [reply prov/sm2/src/sm2.h:186]: done

**Pull request merged**